brave/brave-core
# Merge "DNSLink" and "Redirect gateway-like resouces" settings

Resolves https://github.com/brave/brave-browser/issues/31097

2 settings are merged into a new one, so if any of previous settings were enabled, we switch the new setting on

Sec review : https://github.com/brave/security/issues/1315



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1) Take fresh browser profile
2) Enable "Automatically redirect to IPFS pages via DNSLink when possible" setting
3) Update browser
4) Check that there is a new setting which is DISABLED 
5) Repeat with "Redirect IPFS resources to the configured IPFS gateway" settings
![Uploading image.png…]()
..
6) Enable both DNSLink and "Redirect IPFS resources" and check that new setting is enabled
